packages/next/src/cli/next-dev.ts

```diff
@@ -685,6 +696,32 @@ If you cannot make the changes above, but still want to try out\nNext.js v13 wit
         knownFiles.forEach((_, key) => previousInstrumentationFiles.add(key))
       })
 
+      const configFileWatcher = new Watchpack()
```

Comment:
while you're at it, do you mind refactoring out the watchers logic into separate functions? 

```
runConfigFileWatcher()
runInstrumentationFileWatcher()
runDirectoryChangeWatcher()
```

Reply:
:+1: added.

Also tried to patch a few issues like refreshing other workers after renaming the directory etc. It's still broken in subtle ways, but I think that it should be good enough now.
It's hard to fix those without redoing the whole thing as I tried in #46501